BraveWallet/Crypto/Stores/SwapTokenStore.swift

```diff
@@ -176,9 +176,61 @@ public class SwapTokenStore: ObservableObject {
       completion(nil)
       return
     }
-
-    rpcService.balance(for: token, in: account) { balance in
-      completion(balance.map { BDouble($0) })
+    
+    let balanceFormatter = WeiFormatter(decimalFormatStyle: .decimals(precision: Int(token.decimals)))
```

Comment:
I think this `fetchTokenBalance(for:completion:)` is the same as in `SendTokenStore` except with a completion closure, maybe we should move it to `RpcServiceExtension.swift` with the existing helper?
We might even be able to update the existing helper to include a flag (or `WeiFormatter.DecimalFormatStyle`) 
to determine if we should round or not?